File: SDKs/meraki.com/1.12.0/python/src/sdk/models/operations/getnetworkwirelessairmarshal.py
from dataclasses import dataclass, field
from typing import Any,Optional


@dataclass
class GetNetworkWirelessAirMarshalPathParams:
    network_id: str = field(metadata={'path_param': { 'field_name': 'networkId', 'style': 'simple', 'explode': False }})
    

@dataclass
class GetNetworkWirelessAirMarshalQueryParams:
    t0: Optional[str] = field(default=None, metadata={'query_param': { 'field_name': 't0', 'style': 'form', 'explode': True }})
    timespan: Optional[float] = field(default=None, metadata={'query_param': { 'field_name': 'timespan', 'style': 'form', 'explode': True }})
    

@dataclass
class GetNetworkWirelessAirMarshalRequest:
    path_params: GetNetworkWirelessAirMarshalPathParams = field()
    query_params: GetNetworkWirelessAirMarshalQueryParams = field()
    

@dataclass
class GetNetworkWirelessAirMarshalResponse:
    content_type: str = field()
    status_code: int = field()
    get_network_wireless_air_marshal_200_application_json_object: Optional[dict[str, Any]] = field(default=None)
    

File: SDKs/meraki.com/1.12.0/typescript/src/sdk/models/operations/getnetworkwirelessairmarshal.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class GetNetworkWirelessAirMarshalPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=networkId" })
  networkId: string;
}


export class GetNetworkWirelessAirMarshalQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=t0" })
  t0?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=timespan" })
  timespan?: number;
}


export class GetNetworkWirelessAirMarshalRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetNetworkWirelessAirMarshalPathParams;

  @SpeakeasyMetadata()
  queryParams: GetNetworkWirelessAirMarshalQueryParams;
}


export class GetNetworkWirelessAirMarshalResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  getNetworkWirelessAirMarshal200ApplicationJsonObject?: Map<string, any>;
}
